assets/src/components/activities/custom_dnd/DragCanvas.tsx

```diff
@@ -118,35 +118,43 @@ function focusDraggable(shadowRoot: any, draggableId: string | null, props: Drag
 }
 
 function createTargetDropHandler(shadowRoot: any, props: DragCanvasProps) {
-  return (ev: DragEvent) => {
-    if (
-      ev !== null &&
-      (ev.currentTarget as any).classList.contains('target') &&
-      (ev.currentTarget as any).getAttribute('input_ref') !== null
-    ) {
-      ev.preventDefault();
-      ev.stopPropagation();
-
-      (ev as any).dataTransfer.dropEffect = 'move';
-      const inputVal = (ev as any).dataTransfer.getData('text/plain');
-
-      // bump any existing draggable out of target
-      resetChildDraggables(shadowRoot, ev.currentTarget, props);
-
-      const draggable = getDraggable(shadowRoot, inputVal);
-      //  notify if draggable to be dropped is detaching from another target
-      const prevTargetId = getParentTargetId(draggable);
-      if (prevTargetId) props.onDetach(prevTargetId, inputVal);
-
-      (ev.currentTarget as any).appendChild(draggable);
-      const newlyDropped = (ev.currentTarget as any).children[0];
-      newlyDropped.style.left = '0px';
-      newlyDropped.style.top = '0px';
-      newlyDropped.style.position = 'relative';
-
-      const targetId = (ev.currentTarget as any).getAttribute('input_ref');
-      focusTarget(shadowRoot, targetId, props);
-      props.onSubmitPart(targetId, inputVal);
+  return async (ev: DragEvent) => {
+    try {
+      if (
+        ev !== null &&
+        (ev.currentTarget as any).classList.contains('target') &&
+        (ev.currentTarget as any).getAttribute('input_ref') !== null
+      ) {
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        (ev as any).dataTransfer.dropEffect = 'move';
+        const inputVal = (ev as any).dataTransfer.getData('text/plain');
+
+        // bump any existing draggable out of target
+        resetChildDraggables(shadowRoot, ev.currentTarget, props);
+
+        const draggable = getDraggable(shadowRoot, inputVal);
+        //  notify if draggable to be dropped is detaching from another target
+        const prevTargetId = getParentTargetId(draggable);
+
+        (ev.currentTarget as any).appendChild(draggable);
+        const newlyDropped = (ev.currentTarget as any).children[0];
+        newlyDropped.style.left = '0px';
+        newlyDropped.style.top = '0px';
+        newlyDropped.style.position = 'relative';
+
+        const targetId = (ev.currentTarget as any).getAttribute('input_ref');
+        focusTarget(shadowRoot, targetId, props);
+
+        if (prevTargetId) {
+          await props.onDetach(prevTargetId, inputVal);
```

Comment:
Moving this here and throwing in an await (as well as some upstream changes to make it async) was the actual fix.